smarts/env/gymnasium/hiway_env_v1.py

```diff
@@ -497,3 +504,12 @@ def scenario(self) -> Scenario:
             scenario.Scenario: Current simulated scenario.
         """
         return self._smarts.scenario
+
+    @property
+    def smarts(self):
+        """Gives access to the underlying simulator. Use this carefully.
+
+        Returns:
+            smarts.core.smarts.SMARTS: The smarts simulator instance.
+        """
+        return self._smarts
```

Comment:
Considering we have previously provided `def scenario()`, what is the use case for additionally providing `def smarts()`?

Reply:
Generally just for direct access to SMARTS, I am somewhat cautious about allowing this but if we do not allow it there may be little exposure to actually using SMARTS directly.